src/DataAPI.jl

```diff
@@ -287,4 +287,125 @@ using a `sink` function to materialize the table.
 """
 function allcombinations end
 
+const STYLE_INFO = """
+One of the uses of the metadata `style` is decision
+how the metadata should be propagated when `x` is transformed. This interface
+defines the `:none` style that indicates that metadata should not be propagated
+under transformations. At least this style is allowed by any type
+defining support for metadata.
+"""
+
+const COL_INFO = """
+`col` must have a type that is supported by table `x` for column indexing.
+Following the Tables.jl contract `Symbol` and `Int` are always allowed.
+Passing `col` that is not a column of `x` throws an error.
+"""
+
+"""
+    metadata(x, key::AbstractString; style::Bool=false)
+
+Return metadata value associated with object `x` for key `key`.
+If `x` does not support metadata throw `ArgumentError`.
+If `x` supports metadata, but does not have a mapping for `key` throw
+`KeyError`.
+
+Passing `key` that is not an `AbstractString` to `metadata` throws
+`MethodError`.
+
```

Comment:
I'd drop this as it's to be expected given the signature. Implementations are free to accept more types anyway as long as they document it.
```suggestion
```